debug_module.tex

```diff
@@ -81,6 +82,10 @@ \section{Reset Control} \label{reset}
 operations are reading and writing \RdmDmcontrol. The behavior of other accesses
 is undefined.
 
+A debugger must only use \FdmDmcontrolDmactive to reset the debug module.  If the
```

Comment:
That would sufficient for me, as long as it is clear what the standard mechanisms are.  Is the "global reset signal" referred to in the specification considered standard (albeit not recommended)?

Reply:
Although thinking of this more, what if a poorly implemented watchdog resets the DM because a hart is halted and doesn't kick it in time. Is it hard to ensure that DM just resets normally when it's enabled and some odd reset comes along?

Reply:
I don't exactly follow that comment.  Are you asking how difficult it is to support this poorly implemented watchdog because you think it should be supported if possible?

Reply:
My concern isn't whether the debug module can be reset when it is active (I suspect this isn't hard to implement in hardware).  My concern is about how the entire debug system (including the debugger) is supposed to behave when this happens (as this is now a possibility, with the global system reset).

If the debugger has taken the debug module out of reset (set dmactive to 1), under what circumstances can the debug module spontaneously reset (resulting in dmactive set to 0), and what is the expected behavior of the entire debug system?  Should the debugger continually check if dmactive was spontaneously cleared, and handle this case?  Or is this exceptional behavior that is not allowed, recommended, and/or expected?

Reply:
Generally if there's a reset button on a board, and the user pushes that while the debugger is attached, for some targets that's going to reset the DM. No amount of talking about it in the debug spec will change that, so user-friendly debuggers should be ready for that. But at the same time hardware shouldn't do that (and I think the spec says it shouldn't).

So it's exceptional behavior that isn't really allowed, but is nevertheless expected.

Reply:
I understand you point and agree with it, but it seems to me that the spec is vague enough that it permits not only the scenario that you describe, but many others which don't seem to me as useful to permit.

It seems inconsistent with the requirements of the ndmreset bit in dmcontrol, to permit the system to reset the debug module at any time and under any circumstance.